src/csync/CMakeLists.txt

```diff
@@ -42,11 +33,49 @@ configure_file(csync_version.h.in ${CMAKE_CURRENT_BINARY_DIR}/csync_version.h)
 
 add_library(csync SHARED ${common_SOURCES} ${csync_SRCS})
 
-target_include_directories(
-  csync
-  PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/std
+target_include_directories(csync
+    PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/std
+    # include root source dir in order to make src/common available to the targets
+    PUBLIC ${PROJECT_SOURCE_DIR}/src
+    # make sure auto-generated headers can be included
+    PUBLIC ${PROJECT_BINARY_DIR}
 )
 
+target_compile_definitions(csync
+    PUBLIC -DQT_NO_CAST_TO_ASCII
```

Comment:
Private

Reply:
Which of those? They used to be used in all binaries.

Reply:
Some yes, but some where only set on a directory level.
QT_NO_CAST_TO_ASCII will cause all QString("foo") calls to fail which are still used in the gui code.